src/classify.jl

```diff
@@ -102,15 +102,24 @@ function label(mat, neighborhood = :rook)
     # run through the matrix and make clusters
     for j in axes(mat, 2), i in axes(mat, 1)
         if isfinite(mat[i, j])
-            same = [i - n[1] > 0 && j - n[2] > 0 && mat[i - n[1], j - n[2]] == mat[i, j] for n in neighbors]
```

Comment:
Wouldn't it be sifficient with something like
```
m,k = size(mat)
same = [0 < i - n[1] < m && 0 < j - n[2] > k && mat[i - n[1], j - n[2]] == mat[i, j] for n in neighbors]
```
```

Reply:
we could use the `m` and `k` instead of the axes too

Reply:
@gottacatchenall I think it's a good suggestion - let's do this and I'll merge

Reply:
at some point i tried replacing my fix with the above and it threw a different error, i'll look into it

Reply:
If your fix makes the bug go away, let's go with your fix!